### PR TITLE
uavobjects: conserve stack some and eliminate extra copy

### PR DESCRIPTION
**Pull request by mlyle**
[Review on Reviewable](https://reviewable.io/reviews/d-ronin/dronin/139)

### CODE REVIEW

File: flight/UAVObjects/uavobjecttemplate.c [hunk 1]
```diff
@@ -74,28 +98,10 @@ int32_t $(NAME)Initialize(void)
  */
 void $(NAME)SetDefaults(UAVObjHandle obj, uint16_t instId)
 {
-	$(NAME)Data data;
+	$(NAME)SetDefaultsImpl(obj, instId);
 
-	// Initialize object fields to their default values
-	UAVObjGetInstanceData(obj, instId, &data);
```
Comment (mlyle) [inline flight/UAVObjects/uavobjecttemplate.c:81]: The removal of the UAVObjGetInstanceData is the main change.

Comment (mluessi) [inline flight/UAVObjects/uavobjecttemplate.c:81]: I'm confused by this function. It seems like it sets the UAVO data to all-zero; where are the actual defaults set?

Comment (mlyle) [inline flight/UAVObjects/uavobjecttemplate.c:81]: The UAV object generator inserts code where $(INITFIELDS) is.

Comment (mluessi) [inline flight/UAVObjects/uavobjecttemplate.c:81]: Of course.. it's a template. Silly me.

**Comment by mluessi** [comment]
@mlyle have you tested this? I think it's good to merge on successful test.

**Comment by mlyle** [comment]
No, I still need to smoke test.

**Comment by mlyle** [comment]
Flown on naze32 brushed quad.  Happy.

**Comment by droidicus** [comment]
Tested along with #520, #411, #563, #139 on Sparky2.

Flew, autotuned, etc well.

**Pull request merged**